### ISSUE TITLE
to_csv date_format including hh:mm:ss

### ISSUE THREAD

**Issue by blitzd** [pydata/pandas#10209]
Since 0.16.1 the to_csv output of a dataframe has started including hh:mm:ss with date_format strings like:

date_format='%d-%m-%Y'

Reverting back to 0.16.0 the date output was then formatted properly (without time).

**Comment by jreback** [comment]
You will have to be more specific and show an example. As this is as expected.

```
In [1]: pd.__version__
Out[1]: '0.16.1'

In [2]: DataFrame({'A' : pd.date_range('20130101',periods=5)}).to_csv()
Out[2]: ',A\n0,2013-01-01\n1,2013-01-02\n2,2013-01-03\n3,2013-01-04\n4,2013-01-05\n'

In [3]: DataFrame({'A' : pd.date_range('20130101',periods=5,freq='s')}).to_csv()
Out[3]: ',A\n0,2013-01-01 00:00:00\n1,2013-01-01 00:00:01\n2,2013-01-01 00:00:02\n3,2013-01-01 00:00:03\n4,2013-01-01 00:00:04\n'
```

**Comment by jreback** [comment]
There were some changes to correct a number of bugs in date formatting. This looks like an untested path. Marking as a bug.

**Comment by jreback** [comment]
this is releated to #7791

**Comment by blitzd** [comment]
Actually, date_format appears to be altogether ignored in 0.16.1 to_csv:
```
In [18]: pd.__version__
Out[18]: '0.16.1'

In [19]: pd.DataFrame({'A' : pd.date_range('20130101',periods=5,freq='s')}).to_csv(date_format='%d-%m-%Y')
Out[19]: ',A\n0,2013-01-01 00:00:00\n1,2013-01-01 00:00:01\n2,2013-01-01 00:00:02\n3,2013-01-01 00:00:03\n4,2013-01-01 00:00:04\n'

In [4]: pd.__version__
Out[4]: '0.16.0'

In [5]: pd.DataFrame({'A' : pd.date_range('20130101',periods=5,freq='s')}).to_csv(date_format='%d-%m-%Y')
Out[5]: ',A\n0,01-01-2013\n1,01-01-2013\n2,01-01-2013\n3,01-01-2013\n4,01-01-2013\n'
```

**Comment by jreback** [comment]
closed by #10247
